### ISSUE TITLE
Error: "google-services.json is missing from module root folder" with multiple product flavors

### ISSUE THREAD

**Issue by MiralDesai** [googlesamples/google-services#186]
On Android I keep this error when trying to generate a signed APK in Android Studio. If I run the project directly to my phone (debug build) it works fine. 

I'm using the following version of the plugin in my project build.gradle: 

`classpath 'com.google.gms:google-services:2.0.0-beta6'`

According to the docs (https://developers.google.com/android/guides/google-services-plugin#introduction) since 2.0.0alpha3 there has been support for product flavors in the plugin.

I'm also using gradle-2.10-all.zip `classpath 'com.android.tools.build:gradle:2.0.0-beta6'`

So I'm wondering; Is this a bug in the google-services plugin? 
 
Apologies if this is the wrong place to put this.

**Comment by samtstern** [comment]
@MiralDesai could you provide the relevant parts of your `build.gradle` that define your build types / flavors?

**Comment by MiralDesai** [comment]
Yeah sure. There's not much to it. The package name is the same, the key thing I wanted though is to use different google analytics ids for dev and production, and for other parts of the app that require services like that.

Here is my build.gradle productFlavours part:

```
productFlavors {
        preprod {
            versionCode 1
        }
        demo {
            versionCode 2
        }
        production {
            versionCode 3
        }
    }
```

**Comment by samtstern** [comment]
Ok and where are you putting your JSON files?  Have you tried `src/{preprod,demo,production}/google-services.json`?

**Comment by MiralDesai** [comment]
The JSON files are being put inside the following structure:

```
src/preprod/google-services.json
src/main/google-services.json
src/demo/google-services.json

```

Running a debug build is fine, directly to my phone. The issue seems to be generating a release version.

**Comment by MiralDesai** [comment]
Hmm, just trying the manual XML way and that caused some errors. Instead I added my production google-services.json to the root of app/ and was able to generate a signed APK.

Checking under: 

```
build/generated/res/google-services/flavor/buildType/values/values.xml
build/generated/res/google-services/flavor/buildType/xml/global_tracker.xml

```

I found that all the IDs were correct for my Google Analytics implementation. So preprod and demo was using the ID I wanted them to use and production was using the ID I wanted them to use, so hopefully it works out.

I guess it makes sense, anything in the productFlavor folders override the root files, even if this might be a bug (having to have a root file in the first place).
